### PR TITLE
Change from infura to generic RPC URLS

### PR DESCRIPTION
**Pull request by jferas**
Beginning work for #484

### CODE REVIEW

**Review by alexkeating — COMMENTED** [review]
Code looks good. Left one question!

**Review by garyghayrat — APPROVED** [review]

File: frontend/src/components/models.ts [hunk 1]
```diff
@@ -2,7 +2,15 @@ import { BigNumber } from 'src/utils/ethers';
 import { JsonRpcSigner, Web3Provider } from 'src/utils/ethers';
```
Comment (apbendi) [inline frontend/src/components/models.ts:2]: *arbitrary comment location*

I think we need to make updates to the .env temlplate files in both directories, as well as the README file(s)

**Review by apbendi — COMMENTED** [review]
Hey @jferas, looks good overall. A few small comments, plus I tacked on a commit (as mentioned) for some CNS resolution stuff I found while testing it out.

File: umbra-js/src/classes/Umbra.ts [hunk 2]
```diff
@@ -119,18 +119,18 @@ const isEth = (token: string) => {
...
-  if (chainId === 100) return 'https://rpc.ankr.com/gnosis';
-  if (chainId === 137) return `https://polygon-mainnet.infura.io/v3/${infuraId}`;
-  if (chainId === 42161) return `https://arbitrum-mainnet.infura.io/v3/${infuraId}`;
-  if (chainId === 11155111) return `https://sepolia.infura.io/v3/${infuraId}`;
-  throw new Error(`No Infura URL for chainId ${chainId}.`);
+  if (chainId === 1 || chainId === 1337) return String(process.env.MAINNET_RPC_URL);
+  if (chainId === 10) return String(process.env.OPTIMISM_RPC_URL);
+  if (chainId === 100) return String(process.env.GNOSIS_RPC_URL);
+  if (chainId === 137) return String(process.env.POLYGON_RPC_URL);
+  if (chainId === 42161) return String(process.env.ARBITRUM_RPC_URL);
```
[the remark this answers is not in the document]
Comment (jferas) [reply umbra-js/src/classes/Umbra.ts]: Done: [ed27af7](https://github.com/ScopeLift/umbra-protocol/pull/636/commits/ed27af7129464270f6069e30e7092b1405a80e91)

File: umbra-js/src/classes/Umbra.ts [hunk 3]
```diff
@@ -119,18 +119,18 @@ const isEth = (token: string) => {
...
+  if (chainId === 11155111) return String(process.env.MAINNET_SEPOLIA_URL);
```
[the remark this answers is not in the document]
Comment (jferas) [reply umbra-js/src/classes/Umbra.ts]: Done: [ed27af7](https://github.com/ScopeLift/umbra-protocol/pull/636/commits/ed27af7129464270f6069e30e7092b1405a80e91)

File: frontend/src/components/models.ts (hunk 1 above)
Comment (jferas) [reply frontend/src/components/models.ts:2]: Done: https://github.com/ScopeLift/umbra-protocol/pull/636/commits/edb08fd615f75abf21c03d4c3304582b663534df

**Review by alexkeating — APPROVED** [review]

**Pull request merged**